app/javascript/app/components/table.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+
+function THeader(props) {
+  return (
+    <thead>
+      <tr>
+        {props.config.map((field, i) =>
+          <th key={'header-' + field.header}>
+            {field.header}
+          </th>
+        )}
+      </tr>
+    </thead>
+  );
+}
+
+function Row(props) {
+  return (
+    <tr key={'offer-' + props.offer_id + '-row'}>
+      {props.config.map((field, i) =>
+        <td key={'offer-' + props.offer_id + '-row-' + i}>
+          {field.data(props.offer)}
+        </td>
+      )}
+    </tr>
+  );
+}
+
+class AdminTable extends React.Component {
+  render() {
+    return (
+      <Table striped bordered condensed hover>
+        <THeader config={this.props.config} />
+        <tbody>
+          this.props.offers().map((offer) =>
```

Comment:
- `this.props.offers()` generates an error because it is a List, not a function
- I'm wondering if this was meant to be spread into a key and a value, as referenced in the next line
both `key` and `value` are undefined otherwise
- `this.props.offers` is an Immutable List, but in `config` in ` app/javascript/app/components/admin.js`, you're attempting to access nested elements in the List using JS syntax, not Immutable syntax
I'm not sure whether that was intentional and the individual offers are, in fact, JS objects?